WalletWasabi/WabiSabi/Models/MultipartyTransaction/Signing.cs

```diff
@@ -0,0 +1,88 @@
+using NBitcoin;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using WalletWasabi.WabiSabi.Backend.Models;
+
+namespace WalletWasabi.WabiSabi.Models.MultipartyTransaction
+{
+	public record Signing(Parameters Parameters, ImmutableArray<Coin> Inputs, ImmutableArray<TxOut> Outputs) : State
```

Comment:
What does this class do? It requires a better name.

Reply:
it represents a state in the signing of the transaction, where the only possible action is adding a signature.

Reply:
Now that @nothingmuch mentions it I think it is true. In C# we would call it `SigningState` and `ContructionState` even if we know they are `State` because that's something you can know only if you see the class definition.

Reply:
I meant it should be something like CoinJoinSignatureBuilder.

Reply:
Oh I see. @nothingmuch is modeling this as states and actions that make the transition to a new state while you are saying that whatever builds a transaction is a transaction builder, what makes sense too. I think both visions could be made compatible if the final state gives us a transaction builder.

Reply:
I think the emphasis on the state is more important than the fact that it builds transactions, for both implementation and conceptual reasons, see https://github.com/zkSNACKs/WalletWasabi/pull/5534#discussion_r608685903